### PR TITLE
Dockerfile.olbase: Add Python 3.8 and pip3

### LINKED ISSUE #3417: Get Python 3 installed in the docker images
Subtask of #3416

### Summary of Requirements

- [x] Should be able to run `python3.8 -m pytest` inside a docker image
- [x] The regular service should run using python 2
- [x] The infogami submodule should still be python 2 compatible
    - Idea: Maybe need 2 infogami submodules pinned to different versions?
- [ ] The tests _DO NOT_ need to all pass; but they need to give the same results as Travis on Python 3

### Stakeholders
@cclauss @paulproteus

### LINKED ISSUE #3418: Create flag to run docker image using only python 3
- Subtask of: #3416 
- Depends on: #3417

An environment variable might be a good way to do this; those can be used from the docker-compose file to execute special paths.

### Summary of Requirements

- [x] Way for a developer to `docker-compose up` and get an environment that uses exclusively Python 3 to run everything (infogami, coverstore, solr-updater, etc)
- [x] Way for a developer to `docker-compose up` and get an environment that uses exclusively Python 2 to run everything (infogami, coverstore, solr-updater, etc). This should be the default.
- [x] The website _DOES NOT_ need to work perfectly (maybe at all); but it should be possible to _start_ with everything using python 3

### Stakeholder

@cclauss @paulproteus

### PR DESCRIPTION
**Pull request by cclauss**
Closes #3417
Closes #3418

### CODE REVIEW

**Comment by cdrini** [comment]
@mekarpeles had success installing https://github.com/pyenv/pyenv on dev.openlibrary.org; maybe we could use that?

**Comment by cclauss** [comment]
python2 -m pytest ...
python3 -m pytest ...

File: docker-compose.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ version: "3"
 services:
   web:
     image: oldev:latest
+    environment:
+      - PYTHON3
```
Comment (cclauss) [inline docker-compose.yml:6]: Maybe `PYTHON=python3` instead?

Then we can `$PYTHON -m pytest`

File: docker/ol-docker-start.sh [hunk 2]
```diff
@@ -15,6 +15,16 @@ reindex-solr() {
   done
 }
 
+if [ -z "$PYTHON3" ]; then
```
Comment (cclauss) [inline docker/ol-docker-start.sh:18]:
```suggestion
if ["$PYTHON"=="python3" ]; then
```

File: .python-version [hunk 3]
```diff
@@ -1 +1,2 @@
-2.7.14
+2.7.6
```
Comment (cclauss) [inline .python-version:1]: All versions of Python < 2.7.9 have a known security flaw that should be avoided.  Now that there is guaranteed to _never_ be another release of legacy Python, I see no reason not to pin to [Python 2.7.18](https://www.python.org/downloads/release/python-2718/).

File: docker/Dockerfile.olbase [hunk 4]
```diff
@@ -10,29 +10,48 @@ RUN groupadd --system openlibrary \
...
 
 # Install LTS version of node.js
 RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - \
     && apt-get install -y nodejs
 
+# Install pyenv
+# Python build deps: https://github.com/pyenv/pyenv/wiki/Common-build-problems#prerequisites
+RUN apt-get -qq update && apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+    libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+    xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
```
Comment (cclauss) [inline docker/Dockerfile.olbase:35]:
```suggestion
    xz-utils tk-dev libffi-dev liblzma-dev python-openssl python3-openssl git
```

File: docker/Dockerfile.olbase [hunk 5]
```diff
@@ -10,29 +10,48 @@ RUN groupadd --system openlibrary \
...
+# Install latest pyenv (https://github.com/pyenv/pyenv-installer)
+USER openlibrary
+RUN curl https://pyenv.run | bash && \
+    echo '\nexport PATH="/home/openlibrary/.pyenv/bin:$PATH"\neval "$(pyenv init -)"\neval "$(pyenv virtualenv-init -)"' >> /home/openlibrary/.bashrc
+ENV PYENV_ROOT /home/openlibrary/.pyenv
+ENV PATH $PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
+
+RUN pyenv install 2.7.6
+RUN pyenv install 3.8.3
+RUN pyenv global 2.7.6 3.8.3
```
Comment (cclauss) [inline docker/Dockerfile.olbase:46]: See above on security exposure on such an old version of legacy Python.

**Comment by cclauss** [comment]
I believe that this one is ready to land.  It helped me quickly find the issue in #3581.  The environment variable is elegant and works well.

File: .python-version [hunk 6]
```diff
@@ -1 +1,2 @@
...
+3.8.3
```
Comment (cclauss) [inline .python-version:2]: 3.8.4 ?!?  I know how difficult it is to upgrade things once they are set.

File: docker/Dockerfile.olbase (hunk 5 above)
Comment (cclauss) [inline docker/Dockerfile.olbase:46]: 3.8.4 ??

File: .python-version (hunk 6 above)
Comment (cdrini) [reply .python-version:2]: OH DARN! IT WAS RELEASED YESTERDAY!! :D I actually check to make sure I got the latest version when I wrote _code--_also_ yesterday :P Python3 will only ever exist inside docker, so updating should be easy from now on :)

Comment (cdrini) [reply .python-version:2]: Let's do it in a future PR I think; it takes too long to do these sort of changes and rebuild and test that it's working.

File: docker/Dockerfile.olbase (hunk 5 above)
Comment (cdrini) [reply docker/Dockerfile.olbase:46]: :P Ditto

**Pull request merged**